packages/autorest.python/autorest/codegen/models/model_type.py

```diff
@@ -138,14 +138,15 @@ def get_json_template_representation(
         optional: bool = True,
         client_default_value_declaration: Optional[str] = None,
         description: Optional[str] = None,
+        for_test: Optional[bool] = False,
     ) -> Any:
         if self._created_json_template_representation:
             return "..."  # do this to avoid loop
         self._created_json_template_representation = True
         if self.discriminated_subtypes:
             # we will instead print the discriminated subtypes
             self._created_json_template_representation = False
-            return self.snake_case_name
+            return f'"{self.snake_case_name}"' if for_test else self.snake_case_name
```

Comment:
i think it would be much cleaner to have a separate function for a json template for test use instead of having a `for_test` everywhere. Something like:

```python
def get_json_template(self) // tests will use this one

def get_json_template_representation(self):
  # existing function just calls self.get_json_template
  return f'"{self.get_json_template()}"'
```

Reply:
99% of json template for test and sample are same. The only difference is that json in test have to pass lint check, and that's why I need `for_test` to make some minor change. I also had similar thought at first. Then I find we still need common function for test and sample which supports `for_test`.

Reply:
since they are 99% the same, it makes more sense to split it into 2 operations, because right now you're touching every single method and file and adding more argument clutter. You can have `get_json_template` be the main method with the bulk of the code, and just add `get_json_template_representation` in 2 places

```
# _base.py
def get_json_template(self):
  ...

def get_json_template_representation(self):
  return self.get_json_template()
```

then in 
```
# _model.py

def get_json_template_representation(self):
  return f'"{self.get_json_template()}"'
```

Reply:
Updated